src/app/drive/services/downloadManager.service.ts

```diff
@@ -300,73 +327,96 @@ export class DownloadManagerService {
+            fileId: (driveItem as DriveFileData).fileId,
+            bucketId: (driveItem as DriveFileData).bucket,
+            creds: {
+              user: (driveItem as AdvancedSharedItem).credentials?.networkUser ?? credentials.credentials.user,
+              pass: (driveItem as AdvancedSharedItem).credentials?.networkPass ?? credentials.credentials.pass,
+            },
+            mnemonic: (driveItem as AdvancedSharedItem).credentials?.mnemonic ?? credentials.mnemonic,
+            options: {
+              abortController,
+              notifyProgress: (totalBytes, downloadedBytes) => {
+                const progress = downloadedBytes / totalBytes;
 
-              downloadProgress[index] = progress;
+                downloadProgress[index] = progress;
 
-              updateProgressCallback(calculateProgress());
+                updateProgressCallback(calculateProgress());
+              },
             },
-          },
-        });
+          });
 
-        const sourceBlob = await binaryStreamToBlob(downloadedFileStream);
-        await updateDatabaseFileSourceData({
-          folderId: driveItem.folderId,
-          sourceBlob,
-          fileId: driveItem.id,
-          updatedAt: driveItem.updatedAt,
-        });
+          const sourceBlob = await binaryStreamToBlob(downloadedFileStream);
+          await updateDatabaseFileSourceData({
+            folderId: driveItem.folderId,
+            sourceBlob,
+            fileId: driveItem.id,
+            updatedAt: driveItem.updatedAt,
+          });
 
-        fileStream = sourceBlob.stream();
+          fileStream = sourceBlob.stream();
+        }
+        folderZip.addFile(`${driveItem.name}${driveItem.type ? '.' + driveItem.type : ''}`, fileStream);
+      } catch (error: any) {
+        if (this.isRequiredHandleError(error)) {
+          folderZip.abort();
+          throw error;
+        }
+        failedItems.push(driveItem);
       }
-      folderZip.addFile(`${driveItem.name}${driveItem.type ? '.' + driveItem.type : ''}`, fileStream);
     };
 
     const addFolderToZip = async (index: number, driveItem: DownloadItemType) => {
-      await downloadFolderAsZip({
-        folder: driveItem as DriveFolderData,
-        isSharedFolder: options.areSharedItems,
-        foldersIterator: createFoldersIterator,
-        filesIterator: createFilesIterator,
-        updateProgress: (progress) => {
-          downloadProgress[index] = progress;
-          updateProgressCallback(calculateProgress());
-        },
-        updateNumItems: incrementItemCount,
-        options: {
-          destination: folderZip,
-          closeWhenFinished: false,
-          credentials: {
-            user: (driveItem as AdvancedSharedItem).credentials?.networkUser ?? credentials.credentials.user,
-            pass: (driveItem as AdvancedSharedItem).credentials?.networkPass ?? credentials.credentials.pass,
+      try {
+        const res = await downloadFolderAsZip({
```

Comment:
it would be good to give a more descriptive name to this const, `res` looks like a server response